MapboxCoreNavigation/sequence.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+extension Sequence where Element: Hashable {
+    func compatibleFlatMap<ElementOfResult>(_ transform: (Element) throws -> ElementOfResult?) rethrows -> [ElementOfResult] {
```

Comment:
I'm not crazy about the name here -- is there another way we can define `compactMap` to delegate to `flatMap` where `compactMap` doesn't exist?